Test "TestTribeTaskAgreements" in mgmt/rest/tribe_test.go is flaky.

Spurious failure observed in Travis CI with the Go 1.4.x configuration.

Failed build log follows.

```
Using worker: worker-linux-docker-acf76d55.prod.travis-ci.org:travis-linux-7

travis_fold:start:system_info
[0K[33;1mBuild system information[0m
Build language: go
Build group: stable
Build dist: precise
[34m[1mBuild image provisioning date and time[0m
Thu Feb  5 15:09:33 UTC 2015
[34m[1mOperating System Details[0m
Distributor ID:	Ubuntu
Description:	Ubuntu 12.04.5 LTS
Release:	12.04
Codename:	precise
[34m[1mLinux Version[0m
3.13.0-29-generic
[34m[1mCookbooks Version[0m
a68419e https://github.com/travis-ci/travis-cookbooks/tree/a68419e
[34m[1mGCC version[0m
gcc (Ubuntu/Linaro 4.6.3-1ubuntu5) 4.6.3
Copyright (C) 2011 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[34m[1mLLVM version[0m
clang version 3.4 (tags/RELEASE_34/final)
Target: x86_64-unknown-linux-gnu
Thread model: posix
[34m[1mPre-installed Ruby versions[0m
ruby-1.9.3-p551
[34m[1mPre-installed Node.js versions[0m
v0.10.36
[34m[1mPre-installed Go versions[0m
1.4.1
[34m[1mRedis version[0m
redis-server 2.8.19
[34m[1mriak version[0m
2.0.2
[34m[1mMongoDB version[0m
MongoDB 2.4.12
[34m[1mCouchDB version[0m
couchdb 1.6.1
[34m[1mNeo4j version[0m
1.9.4
[34m[1mRabbitMQ Version[0m
3.4.3
[34m[1mElasticSearch version[0m
1.4.0
[34m[1mInstalled Sphinx versions[0m
2.0.10
2.1.9
2.2.6
[34m[1mDefault Sphinx version[0m
2.2.6
[34m[1mInstalled Firefox version[0m
firefox 31.0esr
[34m[1mPhantomJS version[0m
1.9.8
[34m[1mant -version[0m
Apache Ant(TM) version 1.8.2 compiled on December 3 2011
[34m[1mmvn -version[0m
Apache Maven 3.2.5 (12a6b3acb947671f09b81f49094c53f426d8cea1; 2014-12-14T17:29:23+00:00)
Maven home: /usr/local/maven
Java version: 1.7.0_76, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-7-oracle/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "3.13.0-29-generic", arch: "amd64", family: "unix"
travis_fold:end:system_info
[0K
travis_fold:start:git.checkout
[0Ktravis_time:start:0a5090c0
[0K$ git clone --depth=50 https://github.com/intelsdi-x/snap.git intelsdi-x/snap
Cloning into 'intelsdi-x/snap'...
remote: Counting objects: 1252, done.[K
remote: Compressing objects:   0% (1/667)   [K
remote: Compressing objects:   1% (7/667)   [K
remote: Compressing objects:   2% (14/667)   [K
remote: Compressing objects:   3% (21/667)   [K
remote: Compressing objects:   4% (27/667)   [K
remote: Compressing objects:   5% (34/667)   [K
remote: Compressing objects:   6% (41/667)   [K
remote: Compressing objects:   7% (47/667)   [K
remote: Compressing objects:   8% (54/667)   [K
remote: Compressing objects:   9% (61/667)   [K
remote: Compressing objects:  10% (67/667)   [K
remote: Compressing objects:  11% (74/667)   [K
remote: Compressing objects:  12% (81/667)   [K
remote: Compressing objects:  13% (87/667)   [K
remote: Compressing objects:  14% (94/667)   [K
remote: Compressing objects:  15% (101/667)   [K
remote: Compressing objects:  16% (107/667)   [K
remote: Compressing objects:  17% (114/667)   [K
remote: Compressing objects:  18% (121/667)   [K
remote: Compressing objects:  19% (127/667)   [K
remote: Compressing objects:  20% (134/667)   [K
remote: Compressing objects:  21% (141/667)   [K
remote: Compressing objects:  22% (147/667)   [K
remote: Compressing objects:  23% (154/667)   [K
remote: Compressing objects:  24% (161/667)   [K
remote: Compressing objects:  25% (167/667)   [K
remote: Compressing objects:  26% (174/667)   [K
remote: Compressing objects:  27% (181/667)   [K
remote: Compressing objects:  28% (187/667)   [K
remote: Compressing objects:  29% (194/667)   [K
remote: Compressing objects:  30% (201/667)   [K
remote: Compressing objects:  31% (207/667)   [K
remote: Compressing objects:  32% (214/667)   [K
remote: Compressing objects:  33% (221/667)   [K
remote: Compressing objects:  34% (227/667)   [K
remote: Compressing objects:  35% (234/667)   [K
remote: Compressing objects:  36% (241/667)   [K
remote: Compressing objects:  37% (247/667)   [K
remote: Compressing objects:  38% (254/667)   [K
remote: Compressing objects:  39% (261/667)   [K
remote: Compressing objects:  40% (267/667)   [K
remote: Compressing objects:  41% (274/667)   [K
remote: Compressing objects:  42% (281/667)   [K
remote: Compressing objects:  43% (287/667)   [K
remote: Compressing objects:  44% (294/667)   [K
remote: Compressing objects:  45% (301/667)   [K
remote: Compressing objects:  46% (307/667)   [K
remote: Compressing objects:  47% (314/667)   [K
remote: Compressing objects:  48% (321/667)   [K
remote: Compressing objects:  49% (327/667)   [K
remote: Compressing objects:  50% (334/667)   [K
remote: Compressing objects:  51% (341/667)   [K
remote: Compressing objects:  52% (347/667)   [K
remote: Compressing objects:  53% (354/667)   [K
remote: Compressing objects:  54% (361/667)   [K
remote: Compressing objects:  55% (367/667)   [K
remote: Compressing objects:  56% (374/667)   [K
remote: Compressing objects:  57% (381/667)   [K
remote: Compressing objects:  58% (387/667)   [K
remote: Compressing objects:  59% (394/667)   [K
remote: Compressing objects:  60% (401/667)   [K
remote: Compressing objects:  61% (407/667)   [K
remote: Compressing objects:  62% (414/667)   [K
remote: Compressing objects:  63% (421/667)   [K
remote: Compressing objects:  64% (427/667)   [K
remote: Compressing objects:  65% (434/667)   [K
remote: Compressing objects:  66% (441/667)   [K
remote: Compressing objects:  67% (447/667)   [K
remote: Compressing objects:  68% (454/667)   [K
remote: Compressing objects:  69% (461/667)   [K
remote: Compressing objects:  70% (467/667)   [K
remote: Compressing objects:  71% (474/667)   [K
remote: Compressing objects:  72% (481/667)   [K
remote: Compressing objects:  73% (487/667)   [K
remote: Compressing objects:  74% (494/667)   [K
remote: Compressing objects:  75% (501/667)   [K
remote: Compressing objects:  76% (507/667)   [K
remote: Compressing objects:  77% (514/667)   [K
remote: Compressing objects:  78% (521/667)   [K
remote: Compressing objects:  79% (527/667)   [K
remote: Compressing objects:  80% (534/667)   [K
remote: Compressing objects:  81% (541/667)   [K
remote: Compressing objects:  82% (547/667)   [K
remote: Compressing objects:  83% (554/667)   [K
remote: Compressing objects:  84% (561/667)   [K
remote: Compressing objects:  85% (567/667)   [K
remote: Compressing objects:  86% (574/667)   [K
remote: Compressing objects:  87% (581/667)   [K
remote: Compressing objects:  88% (587/667)   [K
remote: Compressing objects:  89% (594/667)   [K
remote: Compressing objects:  90% (601/667)   [K
remote: Compressing objects:  91% (607/667)   [K
remote: Compressing objects:  92% (614/667)   [K
remote: Compressing objects:  93% (621/667)   [K
remote: Compressing objects:  94% (627/667)   [K
remote: Compressing objects:  95% (634/667)   [K
remote: Compressing objects:  96% (641/667)   [K
remote: Compressing objects:  97% (647/667)   [K
remote: Compressing objects:  98% (654/667)   [K
remote: Compressing objects:  99% (661/667)   [K
remote: Compressing objects: 100% (667/667)   [K
remote: Compressing objects: 100% (667/667), done.[K
Receiving objects:   0% (1/1252)   
Receiving objects:   1% (13/1252)   
Receiving objects:   2% (26/1252)   
Receiving objects:   3% (38/1252)   
Receiving objects:   4% (51/1252)   
Receiving objects:   5% (63/1252)   
Receiving objects:   6% (76/1252)   
Receiving objects:   7% (88/1252)   
Receiving objects:   8% (101/1252)   
Receiving objects:   9% (113/1252)   
Receiving objects:  10% (126/1252)   
Receiving objects:  11% (138/1252)   
Receiving objects:  12% (151/1252)   
Receiving objects:  13% (163/1252)   
Receiving objects:  14% (176/1252)   
Receiving objects:  15% (188/1252)   
Receiving objects:  16% (201/1252)   
Receiving objects:  17% (213/1252)   
Receiving objects:  18% (226/1252)   
Receiving objects:  19% (238/1252)   
Receiving objects:  20% (251/1252)   
Receiving objects:  21% (263/1252)   
Receiving objects:  22% (276/1252)   
Receiving objects:  23% (288/1252)   
Receiving objects:  24% (301/1252)   
Receiving objects:  25% (313/1252)   
Receiving objects:  26% (326/1252)   
Receiving objects:  27% (339/1252)   
Receiving objects:  28% (351/1252)   
Receiving objects:  29% (364/1252)   
Receiving objects:  30% (376/1252)   
Receiving objects:  31% (389/1252)   
Receiving objects:  32% (401/1252)   
Receiving objects:  33% (414/1252)   
Receiving objects:  34% (426/1252)   
Receiving objects:  35% (439/1252)   
Receiving objects:  36% (451/1252)   
Receiving objects:  37% (464/1252)   
Receiving objects:  38% (476/1252)   
Receiving objects:  39% (489/1252)   
Receiving objects:  40% (501/1252)   
Receiving objects:  41% (514/1252)   
Receiving objects:  42% (526/1252)   
Receiving objects:  43% (539/1252)   
Receiving objects:  44% (551/1252)   
Receiving objects:  45% (564/1252)   
Receiving objects:  46% (576/1252)   
Receiving objects:  47% (589/1252)   
Receiving objects:  48% (601/1252)   
Receiving objects:  49% (614/1252)   
Receiving objects:  50% (626/1252)   
Receiving objects:  51% (639/1252)   
Receiving objects:  52% (652/1252)   
Receiving objects:  53% (664/1252)   
Receiving objects:  54% (677/1252)   
Receiving objects:  55% (689/1252)   
Receiving objects:  56% (702/1252)   
Receiving objects:  57% (714/1252)   
Receiving objects:  58% (727/1252)   
Receiving objects:  59% (739/1252)   
Receiving objects:  60% (752/1252)   
Receiving objects:  61% (764/1252)   
Receiving objects:  62% (777/1252)   
Receiving objects:  63% (789/1252)   
Receiving objects:  64% (802/1252)   
Receiving objects:  65% (814/1252)   
Receiving objects:  66% (827/1252)   
Receiving objects:  67% (839/1252)   
Receiving objects:  68% (852/1252)   
Receiving objects:  69% (864/1252)   
Receiving objects:  70% (877/1252)   
Receiving objects:  71% (889/1252)   
Receiving objects:  72% (902/1252)   
Receiving objects:  73% (914/1252)   
Receiving objects:  74% (927/1252)   
Receiving objects:  75% (939/1252)   
Receiving objects:  76% (952/1252)   
Receiving objects:  77% (965/1252)   
Receiving objects:  78% (977/1252)   
Receiving objects:  79% (990/1252)   
remote: Total 1252 (delta 673), reused 998 (delta 547), pack-reused 0[K
Receiving objects:  80% (1002/1252)   
Receiving objects:  81% (1015/1252)   
Receiving objects:  82% (1027/1252)   
Receiving objects:  83% (1040/1252)   
Receiving objects:  84% (1052/1252)   
Receiving objects:  85% (1065/1252)   
Receiving objects:  86% (1077/1252)   
Receiving objects:  87% (1090/1252)   
Receiving objects:  88% (1102/1252)   
Receiving objects:  89% (1115/1252)   
Receiving objects:  90% (1127/1252)   
Receiving objects:  91% (1140/1252)   
Receiving objects:  92% (1152/1252)   
Receiving objects:  93% (1165/1252)   
Receiving objects:  94% (1177/1252)   
Receiving objects:  95% (1190/1252)   
Receiving objects:  96% (1202/1252)   
Receiving objects:  97% (1215/1252)   
Receiving objects:  98% (1227/1252)   
Receiving objects:  99% (1240/1252)   
Receiving objects: 100% (1252/1252)   
Receiving objects: 100% (1252/1252), 545.71 KiB | 0 bytes/s, done.
Resolving deltas:   0% (0/673)   
Resolving deltas:   1% (7/673)   
Resolving deltas:   2% (14/673)   
Resolving deltas:   4% (30/673)   
Resolving deltas:   5% (36/673)   
Resolving deltas:   6% (42/673)   
Resolving deltas:  13% (88/673)   
Resolving deltas:  14% (97/673)   
Resolving deltas:  15% (103/673)   
Resolving deltas:  17% (115/673)   
Resolving deltas:  18% (122/673)   
Resolving deltas:  19% (128/673)   
Resolving deltas:  21% (144/673)   
Resolving deltas:  22% (149/673)   
Resolving deltas:  23% (155/673)   
Resolving deltas:  24% (165/673)   
Resolving deltas:  25% (170/673)   
Resolving deltas:  26% (176/673)   
Resolving deltas:  27% (183/673)   
Resolving deltas:  28% (189/673)   
Resolving deltas:  29% (196/673)   
Resolving deltas:  30% (204/673)   
Resolving deltas:  31% (212/673)   
Resolving deltas:  32% (216/673)   
Resolving deltas:  33% (225/673)   
Resolving deltas:  34% (232/673)   
Resolving deltas:  35% (236/673)   
Resolving deltas:  36% (245/673)   
Resolving deltas:  37% (250/673)   
Resolving deltas:  39% (265/673)   
Resolving deltas:  41% (277/673)   
Resolving deltas:  42% (285/673)   
Resolving deltas:  43% (291/673)   
Resolving deltas:  44% (299/673)   
Resolving deltas:  45% (303/673)   
Resolving deltas:  46% (312/673)   
Resolving deltas:  47% (317/673)   
Resolving deltas:  48% (328/673)   
Resolving deltas:  49% (332/673)   
Resolving deltas:  50% (337/673)   
Resolving deltas:  51% (344/673)   
Resolving deltas:  52% (351/673)   
Resolving deltas:  53% (357/673)   
Resolving deltas:  54% (365/673)   
Resolving deltas:  55% (371/673)   
Resolving deltas:  56% (378/673)   
Resolving deltas:  57% (384/673)   
Resolving deltas:  58% (392/673)   
Resolving deltas:  59% (399/673)   
Resolving deltas:  60% (404/673)   
Resolving deltas:  61% (412/673)   
Resolving deltas:  62% (418/673)   
Resolving deltas:  63% (424/673)   
Resolving deltas:  64% (433/673)   
Resolving deltas:  65% (439/673)   
Resolving deltas:  66% (450/673)   
Resolving deltas:  67% (451/673)   
Resolving deltas:  68% (460/673)   
Resolving deltas:  72% (488/673)   
Resolving deltas:  73% (492/673)   
Resolving deltas:  74% (500/673)   
Resolving deltas:  78% (531/673)   
Resolving deltas:  79% (532/673)   
Resolving deltas:  80% (539/673)   
Resolving deltas:  82% (555/673)   
Resolving deltas:  86% (584/673)   
Resolving deltas:  87% (592/673)   
Resolving deltas:  88% (593/673)   
Resolving deltas:  89% (599/673)   
Resolving deltas:  90% (606/673)   
Resolving deltas:  91% (619/673)   
Resolving deltas:  92% (622/673)   
Resolving deltas:  93% (626/673)   
Resolving deltas:  94% (634/673)   
Resolving deltas:  95% (643/673)   
Resolving deltas:  96% (647/673)   
Resolving deltas:  99% (669/673)   
Resolving deltas: 100% (673/673)   
Resolving deltas: 100% (673/673), done.
Checking connectivity... done.
travis_time:end:0a5090c0:start=1449699133744600896,finish=1449699134257176935,duration=512576039
[0K$ cd intelsdi-x/snap
travis_time:start:0bd8c0c4
[0K$ git fetch origin +refs/pull/620/merge:
remote: Counting objects: 6, done.[K
remote: Compressing objects:  33% (1/3)   [K
remote: Compressing objects:  66% (2/3)   [K
remote: Compressing objects: 100% (3/3)   [K
remote: Compressing objects: 100% (3/3), done.[K
remote: Total 6 (delta 4), reused 4 (delta 3), pack-reused 0[K
Unpacking objects:  16% (1/6)   
Unpacking objects:  33% (2/6)   
Unpacking objects:  50% (3/6)   
Unpacking objects:  66% (4/6)   
Unpacking objects:  83% (5/6)   
Unpacking objects: 100% (6/6)   
Unpacking objects: 100% (6/6), done.
From https://github.com/intelsdi-x/snap
 * branch            refs/pull/620/merge -> FETCH_HEAD
travis_time:end:0bd8c0c4:start=1449699134260815912,finish=1449699134544173507,duration=283357595
[0K$ git checkout -qf FETCH_HEAD
travis_fold:end:git.checkout
[0K
[33;1mThis job is running on container-based infrastructure, which does not allow use of 'sudo', setuid and setguid executables.[0m
[33;1mIf you require sudo, add 'sudo: required' to your .travis.yml[0m
[33;1mSee http://docs.travis-ci.com/user/workers/container-based-infrastructure/ for details.[0m

[33;1mSetting environment variables from .travis.yml[0m
$ export SNAP_SOURCE=/home/travis/gopath/src/github.com/intelsdi-x/snap
$ export SNAP_PATH=/home/travis/gopath/src/github.com/intelsdi-x/snap/build

travis_time:start:0637817a
[0K$ eval "$(gimme 1.4.2)"
go version go1.4.2 linux/amd64
travis_time:end:0637817a:start=1449699137672233530,finish=1449699139910954069,duration=2238720539
[0K$ export GOPATH=$HOME/gopath
$ export PATH=$HOME/gopath/bin:$PATH
$ mkdir -p $HOME/gopath/src/github.com/intelsdi-x/snap
$ rsync -az ${TRAVIS_BUILD_DIR}/ $HOME/gopath/src/github.com/intelsdi-x/snap/
$ export TRAVIS_BUILD_DIR=$HOME/gopath/src/github.com/intelsdi-x/snap
$ cd $HOME/gopath/src/github.com/intelsdi-x/snap
$ gimme version
v0.2.2
$ go version
go version go1.4.2 linux/amd64
travis_fold:start:go.env
[0K$ go env
GOARCH="amd64"
GOBIN=""
GOCHAR="6"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/travis/gopath"
GORACE=""
GOROOT="/home/travis/.gimme/versions/go1.4.2.linux.amd64"
GOTOOLDIR="/home/travis/.gimme/versions/go1.4.2.linux.amd64/pkg/tool/linux_amd64"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0"
CXX="g++"
CGO_ENABLED="1"
travis_fold:end:go.env
[0Ktravis_fold:start:before_install.1
[0Ktravis_time:start:0784804e
[0K$ go get github.com/tools/godep
travis_time:end:0784804e:start=1449699140166499831,finish=1449699141529575641,duration=1363075810
[0Ktravis_fold:end:before_install.1
[0Ktravis_fold:start:before_install.2
[0Ktravis_time:start:0e051749
[0K$ go get github.com/axw/gocov/gocov
travis_time:end:0e051749:start=1449699141533202161,finish=1449699143857457582,duration=2324255421
[0Ktravis_fold:end:before_install.2
[0Ktravis_fold:start:before_install.3
[0Ktravis_time:start:12a7e39a
[0K$ go get github.com/mattn/goveralls
travis_time:end:12a7e39a:start=1449699143861031051,finish=1449699145460730457,duration=1599699406
[0Ktravis_fold:end:before_install.3
[0Ktravis_fold:start:before_install.4
[0Ktravis_time:start:06f60090
[0K$ go get -u github.com/golang/lint/golint
travis_time:end:06f60090:start=1449699145464526033,finish=1449699148009222409,duration=2544696376
[0Ktravis_fold:end:before_install.4
[0Ktravis_fold:start:before_install.5
[0Ktravis_time:start:0021457b
[0K$ go get golang.org/x/tools/cmd/vet
travis_time:end:0021457b:start=1449699148012827428,finish=1449699148725529436,duration=712702008
[0Ktravis_fold:end:before_install.5
[0Ktravis_fold:start:before_install.6
[0Ktravis_time:start:078d34a8
[0K$ go get golang.org/x/tools/cmd/goimports
travis_time:end:078d34a8:start=1449699148729125037,finish=1449699149436352560,duration=707227523
[0Ktravis_fold:end:before_install.6
[0Ktravis_fold:start:before_install.7
[0Ktravis_time:start:1474e2e0
[0K$ go get github.com/smartystreets/goconvey/convey
travis_time:end:1474e2e0:start=1449699149440032036,finish=1449699154030346140,duration=4590314104
[0Ktravis_fold:end:before_install.7
[0Ktravis_fold:start:before_install.8
[0Ktravis_time:start:0e47d644
[0K$ if [ ! -d $SNAP_SOURCE ]; then mkdir -p $HOME/gopath/src/github.com/intelsdi-x; ln -s $TRAVIS_BUILD_DIR $SNAP_SOURCE; fi
travis_time:end:0e47d644:start=1449699154033978369,finish=1449699154036732256,duration=2753887
[0Ktravis_fold:end:before_install.8
[0Ktravis_fold:start:before_install.9
[0Ktravis_time:start:03de74cf
[0K$ go get golang.org/x/tools/cmd/cover
travis_time:end:03de74cf:start=1449699154040276786,finish=1449699154522197181,duration=481920395
[0Ktravis_fold:end:before_install.9
[0Ktravis_fold:start:install.1
[0Ktravis_time:start:13f7bcc8
[0K$ export TMPDIR=$HOME/tmp
travis_time:end:13f7bcc8:start=1449699154525816498,finish=1449699154528660718,duration=2844220
[0Ktravis_fold:end:install.1
[0Ktravis_fold:start:install.2
[0Ktravis_time:start:09c54dfa
[0K$ mkdir -p $TMPDIR
travis_time:end:09c54dfa:start=1449699154532426334,finish=1449699154536012355,duration=3586021
[0Ktravis_fold:end:install.2
[0Ktravis_fold:start:install.3
[0Ktravis_time:start:190074cc
[0K$ cd $SNAP_SOURCE
travis_time:end:190074cc:start=1449699154539627713,finish=1449699154542470338,duration=2842625
[0Ktravis_fold:end:install.3
[0Ktravis_fold:start:install.4
[0Ktravis_time:start:014d07bc
[0K$ make
make deps
make[1]: Entering directory `/home/travis/gopath/src/github.com/intelsdi-x/snap'
bash -c "./scripts/deps.sh"
Checking snap root for deps
Checking snapctl for deps
Checking snap mgmt/rest for deps
make[1]: Leaving directory `/home/travis/gopath/src/github.com/intelsdi-x/snap'
make all
make[1]: Entering directory `/home/travis/gopath/src/github.com/intelsdi-x/snap'
bash -c "./scripts/build.sh /home/travis/gopath/src/github.com/intelsdi-x/snap true"

****  snap build (v0.9.0-beta-99-g3d3db46)  ****

Source Dir = /home/travis/gopath/src/github.com/intelsdi-x/snap
 Building snapd
 Building snapctl
    snapctl => /home/travis/gopath/src/github.com/intelsdi-x/snap/build/bin/snapctl
 Building Plugin(s)
    snap-collector-mock1 => /home/travis/gopath/src/github.com/intelsdi-x/snap/build/plugin/
    snap-collector-mock2 => /home/travis/gopath/src/github.com/intelsdi-x/snap/build/plugin/
    snap-publisher-file => /home/travis/gopath/src/github.com/intelsdi-x/snap/build/plugin/
    snap-processor-passthru => /home/travis/gopath/src/github.com/intelsdi-x/snap/build/plugin/

*******************
make[1]: Leaving directory `/home/travis/gopath/src/github.com/intelsdi-x/snap'
travis_time:end:014d07bc:start=1449699154546194044,finish=1449699173068409192,duration=18522215148
[0Ktravis_fold:end:install.4
[0Ktravis_time:start:28b66f5c
[0K$ make check 2>&1
make test
make[1]: Entering directory `/home/travis/gopath/src/github.com/intelsdi-x/snap'
export SNAP_PATH=`pwd`/build; bash -c "./scripts/test.sh"
gofmt
goimports
go vet
?   	github.com/intelsdi-x/snap	[no test files]
?   	github.com/intelsdi-x/snap/cmd/snapctl	[no test files]
ok  	github.com/intelsdi-x/snap/control	41.999s	coverage: 59.6% of statements
ok  	github.com/intelsdi-x/snap/control/plugin	16.946s	coverage: 70.4% of statements
ok  	github.com/intelsdi-x/snap/control/plugin/client	1.201s	coverage: 42.8% of statements
ok  	github.com/intelsdi-x/snap/control/plugin/cpolicy	0.018s	coverage: 60.8% of statements
?   	github.com/intelsdi-x/snap/control/plugin/encoding	[no test files]
ok  	github.com/intelsdi-x/snap/control/plugin/encrypter	1.967s	coverage: 65.1% of statements
?   	github.com/intelsdi-x/snap/control/routing	[no test files]
ok  	github.com/intelsdi-x/snap/core	0.171s	coverage: 44.7% of statements
ok  	github.com/intelsdi-x/snap/core/cdata	0.010s	coverage: 59.1% of statements
?   	github.com/intelsdi-x/snap/core/control_event	[no test files]
?   	github.com/intelsdi-x/snap/core/ctypes	[no test files]
?   	github.com/intelsdi-x/snap/core/scheduler_event	[no test files]
?   	github.com/intelsdi-x/snap/core/serror	[no test files]
[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0mtime="2015-12-09T22:14:26Z" level=error msg="error enabling task" _block=enable-task _error="Task must be disabled" _module=scheduler task-id=836b86f7-b0f8-480e-a960-d9fc6a58a566 
[32m.[0m[32m.[0m[31m[0m[33m[0m[32m
46 total assertions[0m

2015/12/09 22:14:26 [DEBUG] memberlist: TCP connection from: 127.0.0.1:38672
2015/12/09 22:14:26 [DEBUG] memberlist: Initiating push/pull sync with: 127.0.0.1:55236
2015/12/09 22:14:26 [DEBUG] memberlist: Initiating push/pull sync with: 127.0.0.1:55236
2015/12/09 22:14:26 [DEBUG] memberlist: TCP connection from: 127.0.0.1:38673
2015/12/09 22:14:26 [DEBUG] memberlist: Initiating push/pull sync with: 127.0.0.1:55236
2015/12/09 22:14:26 [DEBUG] memberlist: TCP connection from: 127.0.0.1:38674
2015/12/09 22:14:26 [DEBUG] memberlist: Initiating push/pull sync with: 127.0.0.1:55236
2015/12/09 22:14:26 [DEBUG] memberlist: TCP connection from: 127.0.0.1:38675
[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m[32m.[0m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0mtime="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
[32m.[0m[32m.[0mtime="2015-12-09T22:14:28Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
[32m.[0m[32m.[0m[32m.[0mtime="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
[32m.[0m[32m.[0mtime="2015-12-09T22:14:29Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:29Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:30Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error starting task" _block=start-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:31Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
[32m.[0m2015/12/09 22:14:32 [DEBUG] memberlist: Initiating push/pull sync with: 127.0.0.1:55244
2015/12/09 22:14:32 [DEBUG] memberlist: TCP connection from: 127.0.0.1:33962
[32m.[0m[32m.[0mtime="2015-12-09T22:14:32Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:32Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:32Z" level=error msg="failed to get task state" _block=handleStateQuery _module=tribe _name=member-55245 addr=127.0.0.1 err="Task not found: ID(a1311fb2-cc12-4fc7-8578-2677dd67db48)" msg-id=0907ad06-dbbc-4b25-a0ca-2629d2140237 name=member-55245 port=55246 task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:32Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:33Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
time="2015-12-09T22:14:33Z" level=error msg="error getting task" _block=get-task _error="Task not found" _module=scheduler task-id=a1311fb2-cc12-4fc7-8578-2677dd67db48 
[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33mx[0m[31m[0m[33m
Failures:

  * /home/travis/gopath/src/github.com/intelsdi-x/snap/mgmt/rest/tribe_test.go 
  Line 345:
  Expected: 'false'
  Actual:   'true'
  (Should be equal)

[0m[33m
106 total assertions[0m

--- FAIL: TestTribeTaskAgreements-2 (22.91s)
2015/12/09 22:14:49 [DEBUG] memberlist: Initiating push/pull sync with: 127.0.0.1:55248
2015/12/09 22:14:49 [DEBUG] memberlist: TCP connection from: 127.0.0.1:58871
2015/12/09 22:14:49 [DEBUG] memberlist: Initiating push/pull sync with: 127.0.0.1:55248
2015/12/09 22:14:49 [DEBUG] memberlist: TCP connection from: 127.0.0.1:58872
2015/12/09 22:14:49 [DEBUG] memberlist: Initiating push/pull sync with: 127.0.0.1:55248
2015/12/09 22:14:49 [DEBUG] memberlist: TCP connection from: 127.0.0.1:58873
2015/12/09 22:14:49 [DEBUG] memberlist: TCP connection from: 127.0.0.1:58874
2015/12/09 22:14:49 [DEBUG] memberlist: Initiating push/pull sync with: 127.0.0.1:55248
[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0mtime="2015-12-09T22:14:53Z" level=error msg="plugin not found" _block=handle-remove-plugin _module=tribe _name=member-55247 addr=127.0.0.1 name=member-55247 plugin-name=mock1 plugin-type=collector plugin-version=1 port=55248 
[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[32m[0m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m[0m[33m[0m[33m
218 total assertions[0m

FAIL
coverage: 44.9% of statements
FAIL	github.com/intelsdi-x/snap/mgmt/rest	35.034s
make[1]: *** [test] Error 1
make[1]: Leaving directory `/home/travis/gopath/src/github.com/intelsdi-x/snap'
make: *** [check] Error 2
travis_time:end:28b66f5c:start=1449699173071934725,finish=1449699294211295974,duration=121139361249
[0K
[31;1mThe command "make check 2>&1" exited with 2.[0m

Done. Your build exited with 1.
```

--- a/core/tribe_event/tribe_event.go
+++ b/core/tribe_event/tribe_event.go
@@ -1,0 +1,41 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2015 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tribe_event
+
+import "github.com/intelsdi-x/snap/core"
+
+const (
+	PluginAdded = "Tribe.PluginAdded"
+)
+
+type AddPluginEvent struct {
+	Agreement struct {
+		Name string
+	}
+	Plugin struct {
+		Name    string
+		Type    core.PluginType
+		Version int
+	}
+}
+
+func (e AddPluginEvent) Namespace() string {
+	return PluginAdded
+}

--- a/mgmt/rest/tribe_test.go
+++ b/mgmt/rest/tribe_test.go
@@ -35,8 +35,10 @@ import (
 	log "github.com/Sirupsen/logrus"
 	. "github.com/smartystreets/goconvey/convey"
 
+	"github.com/intelsdi-x/gomit"
 	"github.com/intelsdi-x/snap/control"
 	"github.com/intelsdi-x/snap/core"
+	"github.com/intelsdi-x/snap/core/tribe_event"
 	"github.com/intelsdi-x/snap/mgmt/rest/rbody"
 	"github.com/intelsdi-x/snap/mgmt/tribe"
 	"github.com/intelsdi-x/snap/scheduler"
@@ -151,7 +153,7 @@ func TestTribeTaskAgreements(t *testing.T) {
 	log.SetLevel(log.WarnLevel)
 	numOfNodes := 5
 	aName := "agreement99"
-	mgtPorts, tribePort := startTribes(numOfNodes, "")
+	mgtPorts, tribePort, lpe := startTribes(numOfNodes, "")
 	Convey("A cluster is started", t, func() {
 		Convey("Members are retrieved", func() {
 			for _, i := range mgtPorts {
@@ -206,6 +208,7 @@ func TestTribeTaskAgreements(t *testing.T) {
 						So(resp.Meta.Code, ShouldEqual, 200)
 						So(len(resp.Body.(*rbody.PluginList).LoadedPlugins), ShouldEqual, 1)
 						pluginToUnload := resp.Body.(*rbody.PluginList).LoadedPlugins[0]
+						<-lpe.pluginAddEvent
 						resp = getAgreement(mgtPorts[0], aName)
 						So(resp.Meta.Code, ShouldEqual, 200)
 						So(len(resp.Body.(*rbody.TribeGetAgreement).Agreement.PluginAgreement.Plugins), ShouldEqual, 1)
@@ -278,7 +281,7 @@ func TestTribeTaskAgreements(t *testing.T) {
 									wg.Wait()
 									So(timedOut, ShouldEqual, false)
 									Convey("A new node joins the agreement", func() {
-										mgtPort, _ := startTribes(1, fmt.Sprintf("127.0.0.1:%d", tribePort))
+										mgtPort, _, _ := startTribes(1, fmt.Sprintf("127.0.0.1:%d", tribePort))
 										j := joinAgreement(mgtPort[0], fmt.Sprintf("member-%d", mgtPort[0]), aName)
 										mgtPorts = append(mgtPorts, mgtPort[0])
 										So(j.Meta.Code, ShouldEqual, 200)
@@ -429,7 +432,7 @@ func TestTribePluginAgreements(t *testing.T) {
 	)
 	numOfNodes := 5
 	aName := "agreement1"
-	mgtPorts, _ := startTribes(numOfNodes, "")
+	mgtPorts, _, _ := startTribes(numOfNodes, "")
 	Convey("A cluster is started", t, func() {
 		Convey("Members are retrieved", func() {
 			for _, i := range mgtPorts {
@@ -663,11 +666,29 @@ func TestTribePluginAgreements(t *testing.T) {
 	})
 }
 
+type listenToSeedEvents struct {
+	pluginAddEvent chan struct{}
+}
+
+func newListenToSeedEvents() *listenToSeedEvents {
+	return &listenToSeedEvents{
+		pluginAddEvent: make(chan struct{}),
+	}
+}
+
+func (l *listenToSeedEvents) HandleGomitEvent(e gomit.Event) {
+	switch e.Body.(type) {
+	case *tribe_event.AddPluginEvent:
+		l.pluginAddEvent <- struct{}{}
+	}
+}
+
 // returns an array of the mgtports and the tribe port for the last node
-func startTribes(count int, seed string) ([]int, int) {
+func startTribes(count int, seed string) ([]int, int, *listenToSeedEvents) {
 	var wg sync.WaitGroup
 	var tribePort int
 	var mgtPorts []int
+	lpe := newListenToSeedEvents()
 	for i := 0; i < count; i++ {
 		mgtPort := getAvailablePort()
 		mgtPorts = append(mgtPorts, mgtPort)
@@ -680,12 +701,15 @@ func startTribes(count int, seed string) ([]int, int) {
 		conf.RestAPIPort = mgtPort
 		//conf.MemberlistConfig.PushPullInterval = 5 * time.Second
 		conf.MemberlistConfig.RetransmitMult = conf.MemberlistConfig.RetransmitMult * 2
-		if seed == "" {
-			seed = fmt.Sprintf("%s:%d", "127.0.0.1", tribePort)
-		}
+
 		t, err := tribe.New(conf)
 		if err != nil {
 			panic(err)
+		}
+
+		if seed == "" {
+			seed = fmt.Sprintf("%s:%d", "127.0.0.1", tribePort)
+			t.EventManager.RegisterHandler("tribe.tests", lpe)
 		}
 
 		c := control.New(control.GetDefaultConfig())
@@ -724,7 +748,7 @@ func startTribes(count int, seed string) ([]int, int) {
 		}(mgtPort)
 	}
 	wg.Wait()
-	return mgtPorts, tribePort
+	return mgtPorts, tribePort, lpe
 }
 
 var nextPort uint64 = 55234

--- a/mgmt/tribe/tribe.go
+++ b/mgmt/tribe/tribe.go
@@ -35,6 +35,7 @@ import (
 	"github.com/intelsdi-x/snap/core/control_event"
 	"github.com/intelsdi-x/snap/core/scheduler_event"
 	"github.com/intelsdi-x/snap/core/serror"
+	"github.com/intelsdi-x/snap/core/tribe_event"
 	"github.com/intelsdi-x/snap/mgmt/tribe/agreement"
 	"github.com/intelsdi-x/snap/mgmt/tribe/worker"
 	"github.com/pborman/uuid"
@@ -78,6 +79,7 @@ type tribe struct {
 	taskStateResponses map[string]*taskStateQueryResponse
 	members            map[string]*agreement.Member
 	tags               map[string]string
+	EventManager       *gomit.EventController
 	config             *Config
 
 	pluginCatalog   worker.ManagesPlugins
@@ -117,6 +119,7 @@ func New(cfg *Config) (*tribe, error) {
 		workerQuitChan:  make(chan struct{}),
 		workerWaitGroup: &sync.WaitGroup{},
 		config:          cfg,
+		EventManager:    gomit.NewEventController(),
 	}
 
 	tribe.broadcasts = &memberlist.TransmitLimitedQueue{
@@ -504,6 +507,14 @@ func (t *tribe) AddPlugin(agreementName string, p agreement.Plugin) error {
 		UUID:          uuid.New(),
 		Type:          addPluginMsgType,
 	}
+	defer t.EventManager.Emit(&tribe_event.AddPluginEvent{
+		Agreement: struct{ Name string }{agreementName},
+		Plugin: struct {
+			Name    string
+			Type    core.PluginType
+			Version int
+		}{Name: p.Name(), Type: p.Type_, Version: p.Version_},
+	})
 	if t.handleAddPlugin(msg) {
 		t.broadcast(addPluginMsgType, msg, nil)
 	}

--- a/mgmt/tribe/tribe_test.go
+++ b/mgmt/tribe/tribe_test.go
@@ -841,7 +841,6 @@ func TestTribePluginAgreement(t *testing.T) {
 										}
 									}
 								}(tr)
-
 							}
 							log.Debugf("Waits for %d members of clan1 to have %d plugins\n", numOfTribes, numAddMessages)
 							wg.Wait()
@@ -894,7 +893,6 @@ func TestTribePluginAgreement(t *testing.T) {
 										after := len(t.agreements["clan1"].PluginAgreement.Plugins)
 										So(before, ShouldEqual, after)
 										So(len(t.intentBuffer), ShouldEqual, 1)
-
 										So(len(t.agreements["clan1"].PluginAgreement.Plugins), ShouldBeGreaterThan, numAddMessages)
 										t.handleRemovePlugin(&pluginMsg{
 											LTime:         t.clock.Time(),
@@ -902,17 +900,38 @@ func TestTribePluginAgreement(t *testing.T) {
 											AgreementName: "clan1",
 											Type:          removePluginMsgType,
 										})
-										time.Sleep(50 * time.Millisecond)
 										So(len(t.agreements["clan1"].PluginAgreement.Plugins), ShouldBeGreaterThan, numAddMessages)
 										So(len(t.intentBuffer), ShouldEqual, 0)
+
+										// removes the plugin added to test duplicates
 										t.handleRemovePlugin(&pluginMsg{
 											LTime:         t.clock.Time(),
 											Plugin:        agreement.Plugin{Name_: "pluginABC", Version_: 1},
 											AgreementName: "clan1",
 											Type:          removePluginMsgType,
 										})
-										time.Sleep(50 * time.Millisecond)
 										So(len(t.agreements["clan1"].PluginAgreement.Plugins), ShouldEqual, numAddMessages)
+										// wait for all members of the tribe to get back to 10 plugins
+										wg = sync.WaitGroup{}
+										for _, tr := range tribes {
+											wg.Add(1)
+											go func(tr *tribe) {
+												defer wg.Done()
+												for {
+													select {
+													case <-time.After(1500 * time.Millisecond):
+														c.So(len(t.agreements["clan1"].PluginAgreement.Plugins), ShouldEqual, numAddMessages)
+													default:
+														if clan, ok := tr.agreements["clan1"]; ok {
+															if len(clan.PluginAgreement.Plugins) == numAddMessages {
+																return
+															}
+															time.Sleep(50 * time.Millisecond)
+														}
+													}
+												}
+											}(tr)
+										}
 
 										Convey("Handles a 'remove plugin' messages broadcasted across the cluster", func(c C) {
 											for _, t := range tribes {

